src/kani-compiler/rustc_codegen_kani/src/utils/utils.rs

```diff
@@ -48,6 +50,16 @@ impl<'tcx> GotocCtx<'tcx> {
         Expr::statement_expression(body, t).with_location(loc)
     }
 
+    /// Generates an expression `((dst as usize) % align_of(T) == 0`
+    /// to determine if `dst` is aligned.
+    pub fn is_aligned(&mut self, typ: &'tcx TyS<'_>, dst: Expr) -> Expr {
+        let layout = self.layout_of(typ);
+        let align = Expr::int_constant(layout.align.abi.bytes(), Type::size_t());
+        let cast_dst = dst.cast_to(Type::size_t());
+        let zero = Type::size_t().zero();
+        cast_dst.rem(align).eq(zero)
```

Comment:
What happens for 0 size operations? Does this divide by 0?

```
Note that even if T has size 0, the pointer must be non-null and properly aligned.
```

Reply:
Note that we are using the alignment value, not the size. Alignment is always at least 1: https://doc.rust-lang.org/nomicon/repr-rust.html

Reply:
👍